packages/components/src/components/tablist/tablist.e2e-test.ts

```diff
@@ -0,0 +1,315 @@
+  tab-id="music-tab"
+  aria-controls="music-panel">
+  </mdc-tab>
+  <mdc-tab
+  variant=${variant}
+  ${hideText ? '' : 'text="Documents"'}
+  aria-label="Documents"
+  icon-name="document-bold"
+  tab-id="documents-tab"
+  aria-controls="documents-panel">
+  </mdc-tab>
+  <mdc-tab
+  variant=${variant}
+  ${hideText ? '' : 'text="Meetings"'}
+  aria-label="Meetings"
+  icon-name="calendar-month-bold"
+  tab-id="meetings-tab"
+  aria-controls="meetings-panel">
+  </mdc-tab>`;
+
+const setup = async (args: Set) => {
+  const { componentsPage, ...restArgs } = args;
+  await componentsPage.mount({
+    html: `
+    <div id="mdc-tablist-example">
+      <mdc-tablist
+        active-tab-id=${restArgs['active-tab-id']}
+        data-aria-label=${restArgs['data-aria-label']}
+        data-testid="tablist-component"
+      >
+        ${renderTabs()}
+      </mdc-tablist>
+
+      <!-- The following is an example of the markup for the tab panels.
+      This markup is not part of the component and is only provided for context. -->
+        <div id="calls-panel" role="tabpanel" aria-labelledby="calls-tab" hidden>
+        <p>Calls panel</p>
+      </div>
+      <div id="videos-panel" role="tabpanel" aria-labelledby="videos-tab" hidden>
+        <p>Videos panel</p>
+      </div>
+      <div id="music-panel" role="tabpanel" aria-labelledby="music-tab" hidden>
+        <p>Music panel</p>
+      </div>
+      <div id="documents-panel" role="tabpanel" aria-labelledby="documents-tab" hidden>
+        <p>Documents panel</p>
+      </div>
+      <div id="meetings-panel" role="tabpanel" aria-labelledby="meetings-tab" hidden>
+        <p>Meetings panel</p>
+      </div>
+      <!-- End of example markup for the tab panels -->
+    </div>
+    `,
+    clearDocument: true,
+  });
+
+  const mdcTablist = componentsPage.page.locator('mdc-tablist');
+  await mdcTablist.waitFor();
+
+  return mdcTablist;
+};
+
+test('mdc-tablist', async ({ componentsPage }) => {
+  /**
+   * VISUAL REGRESSION
+   */
+  await test.step('visual-regression', async () => {
+    const stickerSheet = new StickerSheet(componentsPage, 'mdc-tablist');
+
+    // default variant
+    stickerSheet.setChildren(renderTabs());
+    await stickerSheet.createMarkupWithCombination({});
+
+    // variant: glass tab
+    stickerSheet.setChildren(renderTabs(TAB_VARIANTS.GLASS));
+    await stickerSheet.createMarkupWithCombination({});
+
+    // variant: line tab
+    stickerSheet.setChildren(renderTabs(TAB_VARIANTS.LINE));
+    await stickerSheet.createMarkupWithCombination({});
+
+    // variant: icon only tabs
+    stickerSheet.setChildren(renderTabs(TAB_VARIANTS.PILL, true));
+    await stickerSheet.createMarkupWithCombination({});
+
+    await stickerSheet.mountStickerSheet();
+    const container = stickerSheet.getWrapperContainer();
+
+    await test.step('matches screenshot of element', async () => {
+      await componentsPage.visualRegression.takeScreenshot('mdc-tablist', { element: container });
```

Comment:
```suggestion
            await componentsPage.visualRegression.takeScreenshot('mdc-tablist', {
        element: container,
        assertionAfterSwitchingDirection: async (page) => {
          await page.evaluate(() => {
            const tablists = Array.from(document.querySelectorAll('mdc-tablist') || []);

            tablists.forEach((tablist, index) => {
              const tabs = Array.from(tablist.querySelectorAll('mdc-tab')) as Tab[];
              tabs[0]?.click();
              tabs[1]?.click();
            });
          });
        },
```

The `assertionAfterSwitchingDirection ` call is needed because the arrows only re-render when the component state changes.
In an ideal scenario, the direction of the page is not changed on the fly. It is usually preset.
However in this case we need to force a re-render to get the correct arrow icon to show. Otherwise there will be flipped arrow buttons.